deploy/templates/nstemplatetiers/clusterresources/advanced.yaml

```diff
@@ -0,0 +1,25 @@
+apiVersion: template.openshift.io/v1
```

Comment:
yes, we could try something like that, indeed. Let me think about it

Reply:
or maybe we could do something like this:
```
/templates/tiers/advanced
    /namespaces
        /code.yaml
        /dev.yaml
        /stage.yaml
    /cluster
        /cluster.yaml
/templates/tiers/basic
    /namespaces
        /code.yaml
        /dev.yaml
        /stage.yaml
    /cluster
        /cluster.yaml
/templates/tiers/team
    /namespaces
        /code.yaml
        /dev.yaml
    /cluster
        /cluster.yaml
...
```
i.e., we keep the intermediate `namespaces` and `cluster` folders, so we know how to process the files and bind them to the `spec.namespaces` vs `spec.clusterResources` fields in the NSTemplateTier CRs?

Reply:
@alexeykazakov otherwise, we could keep your suggestion but rely on a (strict) file naming scheme to bind the templates to the spec.namespaces vs spec.clusterResources fields in the NSTemplateTier CRs.

Reply:
I would rather stay with what we have now or choose the Alexey's proposal. The additional directories for namespace and cluster templates under separated dirs for every tier looks really confusing and chaotic for me.

Reply:
> The additional directories for namespace and cluster templates under separated dirs for every tier looks really confusing and chaotic for me.
"chaotic", really? 😳
The problem with having all files in the same directory is how to distinguish the namespace-related templates vs the other cluster-scoped templates. For that, we could use a name pattern (eg: all namespace template filenames would have to start with `ns-`). Would it make it less confusing to you @MatousJobanek ?

Reply:
I just thought that having all the (namespace) templates in one big dir is a bit messy. Especially when we have many tiers.
Ideally one dir per tier (and all the tier templates in that dir) would look better IMO.
And yes, for that we would need to use strict file name patterns to distinguish between cluster and ns templates. Which is fine.
Separate folders for cluster and ns would work too for me.
But let’s not spend too much time on it! I’m ok to keep it as is for now and refactor it later if it takes more than couple of hours for you to re-structure the dirs now :)

Reply:
> For that, we could use a name pattern (eg: all namespace template filenames would have to start with ns-). Would it make it less confusing to you

yes, the name pattert would be fine, but I agree with Alexey:

> But let’s not spend too much time on it! I’m ok to keep it as is for now and refactor it later if it takes more than couple of hours for you to re-structure the dirs now :)

so keep it as it is and if we onboard more tiers, then we can think about seperating them :-)